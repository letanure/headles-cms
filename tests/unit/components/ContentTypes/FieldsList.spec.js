import { shallowMount } from '@vue/test-utils'
import Component from '@/components/ContentTypes/FieldsList.vue'
require('babel-plugin-require-context-hook/register')()

describe(`${Component.name}.vue`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Component, {
      propsData: {
        value: [
          { name: 'A', type: 'text' },
          { name: 'B', type: 'radio' },
        ],
      },
      mocks: {
        $t: (msg) => msg,
      },
    })
  })

  describe('smoke tests', () => {
    it('should render', () => expect(wrapper.exists()).toBe(true))
    it('matches snapshot', () => expect(wrapper.element).toMatchSnapshot())
  })
})
