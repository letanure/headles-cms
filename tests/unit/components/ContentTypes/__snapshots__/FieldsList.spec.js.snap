// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldsList.vue smoke tests matches snapshot 1`] = `
<div
  class="FieldsList"
>
  <transition-group-stub
    mode="out-in"
    name="flip-list"
    tag="div"
  >
    <el-card-stub>
      <el-row-stub
        align="top"
        justify="start"
        tag="div"
      >
        <el-col-stub
          span="20"
          tag="div"
        >
          <selectfieldtype-stub
            value="text"
          />
        </el-col-stub>
        <el-col-stub
          span="4"
          tag="div"
        >
          <el-button-stub
            icon=""
            nativetype="button"
            size="small"
            type="danger"
          >
            <i
              class="el-icon-delete"
            />
            <span>
              general.actions.delete.label
            </span>
          </el-button-stub>
        </el-col-stub>
      </el-row-stub>
    </el-card-stub>
    <el-card-stub>
      <el-row-stub
        align="top"
        justify="start"
        tag="div"
      >
        <el-col-stub
          span="20"
          tag="div"
        >
          <selectfieldtype-stub
            value="radio"
          />
        </el-col-stub>
        <el-col-stub
          span="4"
          tag="div"
        >
          <el-button-stub
            icon=""
            nativetype="button"
            size="small"
            type="danger"
          >
            <i
              class="el-icon-delete"
            />
            <span>
              general.actions.delete.label
            </span>
          </el-button-stub>
        </el-col-stub>
      </el-row-stub>
    </el-card-stub>
  </transition-group-stub>
  <el-button-stub
    icon=""
    nativetype="button"
    type="default"
  >
    general.actions.add.label
  </el-button-stub>
</div>
`;
